tests/integration_tests/bugs/test_lp1897099.py

```diff
@@ -0,0 +1,30 @@
+""" Integration test for LP #187099
+
+Ensure that if fallocate fails during mkswap that we fall back to using dd
+
+https://bugs.launchpad.net/cloud-init/+bug/1897099
+"""
+
+import pytest
+
+
+USER_DATA = """\
+#cloud-config
+bootcmd:
+  - echo 'whoops' > /usr/bin/fallocate
+swap:
+  filename: /swap.img
+  size: 10000000
+  maxsize: 10000000
+"""
+
+
+@pytest.mark.user_data(USER_DATA)
+@pytest.mark.no_container
```

Comment:
What do you think to having a more generic skip mechanism which enforces a reason? As a strawman:

```py
@pytest.mark.skip_on_platform("lxd_container", reason="Containers cannot configure swap")
```

I'm concerned that we'll end up with tests that have platform specifications that we don't really understand, which will become a maintenance issue.

Reply:
I'm not sure yet. For tests that require a specific platform, we have the platform specific marks. Besides containers, I don't foresee many tests that will work in multiple VMs/clouds, but not a specific other one. I added a reason as an arg to the `no_container` mark on this test. It's not used anywhere and won't affect anything, but we can always add some restriction in the future that we have to provide a reason. Does that sound ok?

Reply:
Yeah, this LGTM, thanks!